clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -779,6 +781,15 @@ class SingleDeviceFunctionTracker {
     if (!CurrentDecl)
       return;
 
+    /// This will print a uniqued call graph of functions being complied for the
```

Comment:
```suggestion
    /// This will print a uniqued call graph of functions being compiled for the
```
Can you clarify the sentence about what you mean with `uniqued` here?

Reply:
By uniqued I meant that every function is printed only once even if it is called from many placed this is in place to prevent printing way too much. so it will print 1 call path by which a function was called. for example if the real call graph was:
```
A
  B
    C
      ...
  D
    C
      ...
```
this function will print:
```
A
  B
    C
      ...
  D
```

Reply:
Strange... The problem is that I have no idea then that D is calling C.
So perhaps "D C" is better, displaying the second time C but no C children, because this is déjà vu. But the fact that D call C is not déjà vu.

Reply:
it is way too noisy to be useful or performant when there isn't any uniquing. and uniquing doesn't prevent to use this for what I usually use it for, identifying why some code was generated for device.